### PR TITLE
Drop legacy/hack support of OR junction in anything else than legacy Model::addCondition()

### PR DESCRIPTION
**Pull request by mvorisek**

### CODE REVIEW

File: src/Model.php [hunk 1]
```diff
@@ -965,6 +965,13 @@ public function _unset(string $name)
      */
     public function addCondition($field, $operator = null, $value = null)
     {
+        // legacy OR support before Scope was introduced
+        if (func_num_args() === 1 && is_array($field) && count($field) === 1 && is_array(reset($field))) {
```
Comment (mvorisek) [inline src/Model.php:969]: we should probably drop the support later, now fully BC

File: src/Model/Scope/CompoundCondition.php [hunk 2]
```diff
@@ -47,13 +47,7 @@ public function __construct(array $nestedConditions = [], string $junction = sel
...
                     $nestedCondition = [$nestedCondition];
                 }
-
-                // array of OR nested conditions
-                if (count($nestedCondition) === 1 && is_array(reset($nestedCondition))) {
-                    $condition = new self(reset($nestedCondition), self::OR);
-                } else {
-                    $condition = new Condition(...$nestedCondition);
-                }
+                $condition = new Condition(...$nestedCondition);
```
Comment (georgehristov) [inline src/Model/Scope/CompoundCondition.php:50]: I would use a on-liner here:
`$condition = new Condition(...(array) $nestedCondition);`

File: src/Model.php (hunk 1 above)
Comment (georgehristov) [reply src/Model.php:969]: 👍

File: src/Model/Scope/CompoundCondition.php (hunk 2 above)
Comment (mvorisek) [reply src/Model/Scope/CompoundCondition.php:50]: I prefer to keep this is easy to understand nesting levels

The non covered/buggy cases come mostly from bad coding style ;-)

And could be possibly removed in PHP 10 :)

**Review by georgehristov — APPROVED** [review]

File: src/Model/Scope/CompoundCondition.php (hunk 2 above)
Comment (georgehristov) [reply src/Model/Scope/CompoundCondition.php:50]: Please ...

Comment (mvorisek) [reply src/Model/Scope/CompoundCondition.php:50]: thank you :), never

**Pull request merged**